src/helpers/mimetype.ts

```diff
@@ -1,25 +1,36 @@
 import {fromBuffer} from "file-type";
 
-export async function isAllowedMimeType(
+export function isAllowedMimeType(
   file: Buffer,
   allowedMimeTypes: string[]
 ): Promise<boolean> {
-  const fileType = (await fromBuffer(file)) || {mime: "text/plain"};
-  if (fileType) {
-    for (const type of allowedMimeTypes) {
-      const match = fileType.mime.match(type);
-      if (match) {
-        return true;
-      }
-    }
-  }
-  if (allowedMimeTypes.includes("application/json")) {
-    try {
-      JSON.parse(file.toString());
-      return true;
-    } catch (e) {
-      // ignore
-    }
-  }
-  return false;
+  return new Promise<boolean>((resolve) => {
+    fromBuffer(file)
+      .then((fileType) => {
+        let fileTypeWithDefault = fileType || {mime: "text/plain"};
+        if (fileTypeWithDefault) {
+          for (const type of allowedMimeTypes) {
+            const match = fileTypeWithDefault.mime.match(type);
+            if (match) {
+              resolve(true);
+              return;
+            }
+          }
+        }
+        if (allowedMimeTypes.includes("application/json")) {
+          try {
+            JSON.parse(file.toString());
+            resolve(true);
+            return;
+          } catch (e) {
+            // ignore
+          }
+        }
+        resolve(false);
+      })
+      .catch(() => {
+        // return false in case of an error
+        resolve(false);
+      });
+  });
```

Comment:
Even a bit simpler ;-)
Promise then/catch clauses already handle returned values as promise resolved values.

```suggestion
  return fromBuffer(file)
      .then((fileType) => {
        let fileTypeWithDefault = fileType || {mime: "text/plain"};
        if (fileTypeWithDefault) {
          for (const type of allowedMimeTypes) {
            const match = fileTypeWithDefault.mime.match(type);
            if (match) {
              return true;
            }
          }
        }
        if (allowedMimeTypes.includes("application/json")) {
          try {
            JSON.parse(file.toString());
            return true;
          } catch (e) {
            // ignore
          }
        }
        return false;
      })
      .catch(() => false);
```